designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Sections.kt

```diff
@@ -0,0 +1,333 @@
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing.Spacing24
+import org.hisp.dhis.mobile.ui.designsystem.theme.SurfaceColor
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+
+enum class SectionState {
+    OPEN, CLOSE, FIXED;
+
+    fun getNextState(): SectionState {
+        return when (this) {
+            OPEN -> CLOSE
+            CLOSE -> OPEN
+            FIXED -> FIXED
+        }
+    }
+}
+
+@Composable
+fun Section(
+    modifier: Modifier = Modifier,
+    isLastSection: Boolean = false,
+    title: String,
+    description: String?,
+    completedFields: Int,
+    totalFields: Int,
+    state: SectionState,
+    errorCount: Int,
+    warningCount: Int,
+    onNextSection: () -> Unit,
+    content: @Composable ColumnScope.() -> Unit,
+) {
+    var sectionState by remember(state) { mutableStateOf(state) }
+    val bottomPadding = when (state) {
+        SectionState.FIXED -> Spacing.Spacing40
+        else -> Spacing16
+    }
+
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .background(Color.White)
+            .run {
+                if (state != SectionState.FIXED) {
+                    bottomBorder(1.dp, Ash600)
+                } else {
+                    this
+                }
+            }
+            .padding(top = Spacing.Spacing8, bottom = bottomPadding),
+        verticalArrangement = spacedBy(Spacing24),
+    ) {
+        SectionHeader(
+            title = title,
+            description = description,
+            completedFields = completedFields,
+            totalFields = totalFields,
+            sectionState = sectionState,
+            errorCount = errorCount,
+            warningCount = warningCount,
+            onSectionClick = {
+                sectionState = sectionState.getNextState()
+            },
+        )
+        AnimatedVisibility(
+            visible = sectionState != SectionState.CLOSE,
+            enter = expandVertically(expandFrom = Alignment.CenterVertically),
+            exit = shrinkVertically(shrinkTowards = Alignment.CenterVertically),
+        ) {
+            Column(
+                modifier = Modifier
+                    .testTag(SectionTestTag.CONTENT)
+                    .fillMaxWidth(),
+                verticalArrangement = spacedBy(Spacing16),
+            ) {
+                content()
+            }
+        }
+        if (!isLastSection && sectionState == SectionState.OPEN) {
+            Button(
+                modifier = Modifier.align(Alignment.End),
+                style = ButtonStyle.TEXT,
+                text = provideStringResource("action_next"),
+                icon = {
+                    Icon(
+                        imageVector = Icons.Filled.ArrowForward,
+                        contentDescription = "Icon Button",
+                    )
+                },
+                onClick = onNextSection,
+            )
+        }
+    }
+}
+
+@Composable
+fun SectionHeader(
```

Comment:
Maybe this should be private or internal?

Reply:
I've decided to have two different public components to test how the form behaves. If the Section one has good performance we could make SectionHeader private. If not, we will use SectionHeader in the form but Section could be used by other devs.